stdlib/websub/src/main/java/org/ballerinalang/net/websub/BallerinaWebSubConnectorListener.java

```diff
@@ -186,30 +183,39 @@ protected void extractPropertiesAndStartResourceExecution(HttpCarbonMessage http
             response.addHttpContent(new DefaultLastHttpContent());
             HttpUtil.sendOutboundResponse(httpCarbonMessage, response);
 
-            signatureParams[0] = createNotificationStruct(balResource, httpRequest);
+            signatureParams[paramIndex++] = createNotification(httpRequest);
+            signatureParams[paramIndex++] = true;
             if (!RESOURCE_NAME_ON_NOTIFICATION.equals(balResource.getName())) {
-                signatureParams[1] = createCustomNotificationStruct(httpCarbonMessage, balResource, httpRequest);
+                signatureParams[paramIndex++] = createCustomNotification(httpCarbonMessage, balResource, httpRequest);
+                signatureParams[paramIndex] = true;
             }
         }
 
         CallableUnitCallback callback = new WebSubEmptyCallableUnitCallback();
         //TODO handle BallerinaConnectorException
-        Executor.submit(balResource, callback, null, null, signatureParams);
+        ObjectValue service = httpResource.getParentService().getBalService();
+        Executor.submit(scheduler, service, balResource.getName(), callback, null,
+                        signatureParams);
     }
 
     @SuppressWarnings("unchecked")
     private void validateSignature(HttpCarbonMessage httpCarbonMessage, HttpResource httpResource,
-                                   BMap<String, BValue> requestStruct) {
+                                   ObjectValue request) {
         //invoke processWebSubNotification function
-        PackageInfo packageInfo = context.getProgramFile().getPackageInfo(WEBSUB_PACKAGE);
-        FunctionInfo functionInfo = packageInfo.getFunctionInfo("processWebSubNotification");
-        BValue[] returnValues = BVMExecutor.executeFunction(functionInfo.getPackageInfo().getProgramFile(),
-                functionInfo, requestStruct, httpResource.getBalResource().getService().getBValue());
-
-        BError errorStruct = (BError) returnValues[0];
-        if (errorStruct != null) {
+        Object returnValue;
+        try {
+            Object[] args = {request, httpResource.getParentService().getBalService()};
+            returnValue = Executor.executeFunction(scheduler, this.getClass().getClassLoader(), WEBSUB, "commons",
+                                                   "processWebSubNotification", args);
+        } catch (BallerinaException e) {
+            log.debug("Signature Validation failed: " + e.getMessage());
+            httpCarbonMessage.setHttpStatusCode(404);
+            throw WebSubUtils.createError(e.getMessage());
```

Comment:
When would this fail? If there is an issue with the scheduling/function execution is it?

Also at L220 we seem to be throwing a BallerinaException. In both cases we only need to respond with a 404 right?

Reply:
Yes, it catches the exceptions which occurs due to ballerina method invocation failure. The message of exception provides a sufficient detail which is being concatenated here.

The thrown exception at L213 will get caught at L127 and leads it to the failure path which is the 404 error response.